srml/support/src/traits.rs

```diff
@@ -230,7 +230,7 @@ impl<
 /// Abstraction over a fungible assets system.
 pub trait Currency<AccountId> {
 	/// The balance of an account.
-	type Balance: SimpleArithmetic + As<usize> + As<u64> + Codec + Copy + MaybeSerializeDebug + Default;
+	type Balance: SimpleArithmetic + Codec + Copy + MaybeSerializeDebug + Default;
```

Comment:
it seems to me that only having `From<u64>` and `TryInto<u64>` on Currency could be interesting (this would assume that a currency is overset of u64, but this is legit to me)

but SimpleArithmetic bouds `As<u64>` at the moment, this removal only actually removes `As<usize>`

Reply:
Also agree with the concept of `TryInto<...>` (or maybe even `TryFrom<...>`) instead of the current one. Prevents any API break and seems like a generic-enough approach. 

Question though: `Balance` is `SimpleArithmetic` and `SimpleArithmetic` is already `As<u64>`. Isn't this already indirectly implying that `Balance` is still bounded by `As<u64>`?

Reply:
> Question though: Balance is SimpleArithmetic and SimpleArithmetic is already As<u64>. Isn't this already indirectly implying that Balance is still bounded by As<u64>?

Yes, that's what I tried to say in my second paragraph

Reply:
Right, I went blind for a sec apparently. 

@xlc Did you ever test to see if this PR, with its current state, actually solve your initial problem (storing some kind of `Vec` as Balance)? I assume no.

Reply:
No. As other pointed out, `Vec` is not `Copy`able so I need to do it differently.
I was thinking to do it differently, e.g. `[Balance; 4]`. But regardless the approach, it will likely not able to convert into a number and requires this PR.